File: src/app/service/rest/TournamentPatchRequest.ts
import {PlayerDTO} from '../../model/dto/PlayerDTO';
import {ProtocolDTO} from '../../model/dto/ProtocolDTO';
import {TournamentDTO} from '../../model/dto/TournamentDTO';

export interface TournamentPatchRequest {
  tid: number;
  update: {
    tournament?: TournamentDTO;
    players?: PlayerDTO[];
    protocols?: ProtocolDTO[];
  };
}

File: src/main/java/com/oliver/thelper/model/TournamentPatchRequest.java
package com.oliver.thelper.model;

import java.util.List;

public class TournamentPatchRequest {
  public int tid; // cannot be null, intentionally primitive
  public PatchBucket update;

  public static class PatchBucket {
    public Tournament tournament;
    public List<Player> players;
    public List<Protocol> protocols;
  }
}
